sparkql/schema_merger.py

```diff
@@ -53,6 +51,16 @@ class _SchemaMerger:
     See `merge_schemas` for detailed behaviour description.
     """
 
+    @classmethod
+    def __validate_mergeable(cls, data_type: DataType) -> MergeableSparkDataType:
+        if not isinstance(data_type, get_args(MergeableSparkDataType)):
+            raise ValueError(
+                "Data type is not mergeable, expected one of: "
+                f"{[t.__name__ for t in get_args(MergeableSparkDataType)]} "
+                f"but got '{type(data_type).__name__}'"
+            )
+        return cast(MergeableSparkDataType, data_type)
```

Comment:
So, usually, mypy can understand checks using `isinstance` and not require a cast, but here because it's a type union, you cannot use `isinstance(data_type, MergeableSparkDataType)` (my understanding is that it doesn't work for the runtime based on the errors it produce), this is kind of the explanation I found: https://stackoverflow.com/a/70347629/6218726